lib/auth/auth.go

```diff
@@ -313,27 +303,78 @@ func (s *AuthServer) GenerateServerKeys(hostID string, role teleport.Role) (*Pac
 	}, nil
 }
 
+// ValidteToken takes a provisioning token value and finds if it's valid. Returns
+// a list of roles this token allows its owner to assume, or an error if the token
+// cannot be found
+func (s *AuthServer) ValidateToken(token string) (roles teleport.Roles, e error) {
+	// look at static tokesn first:
+	for _, st := range s.StaticTokens {
+		if st.Value == token {
+			return st.Roles, nil
+		}
+	}
+	// look at the tokens in the token storage
+	tok, err := s.Provisioner.GetToken(token)
+	if err != nil {
+		log.Warn(err)
+		return nil, trace.Errorf("token not recognized")
```

Comment:
I have been using teleport in real life and this behavior is user-facing and required. All my changes to errors and logging were made to improve Teleport Administrator experience, namely:

- CLI error messages
- Log output

In both cases I was trying to make it obvious to a user to diagnose and fix configuration errors.